floss/language/go/extract.py

```diff
@@ -40,6 +43,324 @@ def extract_stackstrings(extract_stackstring_pattern, section_data, min_length)
+    """
+    image_base = pe.OPTIONAL_HEADER.ImageBase
+
+    # cache the section data so that we can avoid pefile overhead
+    section_datas: List[Tuple[VA, VA, bytes]] = []
+    for section in pe.sections:
+        if not section.IMAGE_SCN_MEM_READ:
+            continue
+
+        section_datas.append(
+            (image_base + section.VirtualAddress, 
+             image_base + section.VirtualAddress + section.SizeOfRawData,
+             section.get_data())
+        )
+
+    range_votes = collections.Counter()
+    for instance in get_struct_string_instances(pe):
+        section_start, _, section_data = next(filter(lambda s: s[0] <= instance.address < s[1], section_datas))
+
+        instance_offset = instance.address - section_start
+
+        next_null = section_data.find(b"\x00\x00", instance_offset)
+        if next_null == -1:
+            continue
+
+        prev_null = section_data.rfind(b"\x00\x00", 0, instance_offset)
+        if prev_null == -1:
+            continue
+
+        range_votes[(section_start + prev_null, section_start + next_null)] += 1
+
+    for (prev_null, next_null), count in range_votes.most_common():
+        if count == 1:
+            continue
+
+        logger.debug("range vote: 0x%x 0x%x 0x%x", prev_null + section_start, next_null + section_start, count)
+
+    most_common, count = range_votes.most_common(1)[0]
+    return most_common
+
+
+def get_string_blob_strings(pe: pefile.PE) -> Tuple[VA, str]:
+    image_base = pe.OPTIONAL_HEADER.ImageBase
+   
+    string_blob_range = get_string_blob_range(pe)
+    string_blob_start, string_blob_end = string_blob_range
+    string_blob_size = string_blob_end - string_blob_start
+    string_blob_buf = pe.get_data(string_blob_range[0] - image_base, string_blob_size)
+
+    # initialized with pointers to the beginning and end,
+    # so that when we split by pairs, we can get those bookends.
+    string_blob_pointers: List[VA] = [string_blob_range[0], string_blob_range[1]]
+
+    for instance in get_struct_string_instances(pe):
+        if not (string_blob_range[0] <= instance.address < string_blob_range[1]):
+            continue
+
+        string_blob_pointers.append(instance.address)
+
+    for xref in get_lea_xrefs(pe):
+        if not (string_blob_range[0] <= xref < string_blob_range[1]):
+            continue
+
+        string_blob_pointers.append(xref)
+
+    last_size = 0
+    string_blob_pointers = list(sorted(set(string_blob_pointers)))
+    for start, end in zip(string_blob_pointers, string_blob_pointers[1:]):
+        size = end - start
+
+        if not (string_blob_start <= start < string_blob_end):
+            continue
+
+        if not (string_blob_start <= end < string_blob_end):
+            continue
+
+        string_blob_offset = start - string_blob_start
+        sbuf = string_blob_buf[string_blob_offset : string_blob_offset + size]
+
+        try:
+            s = sbuf.decode("utf-8")
+        except UnicodeDecodeError:
+            continue
```

Comment:
TODO: better handle the last string in the string blob range which might have to deal with trailing junk bytes. we could keep trimming bytes off the right hand side until the UTF-8 decodes correctly. we also know the min length from the prior string in the blob (assuming strings are sorted shortest to longest)